Sources/App/Views/ViewModelQueries/PackageShowViewModel.swift

```diff
@@ -0,0 +1,49 @@
+import Fluent
+import Foundation
+import Vapor
+
+
+extension PackageShowView.Model {
+    static func query(database: Database, packageId: Package.Id) -> EventLoopFuture<Self> {
+        Package.query(on: database)
+            .with(\.$repositories)
+            .with(\.$versions) { $0.with(\.$products) }
+            .filter(\.$id == packageId)
+            .first()
+            .unwrap(or: Abort(.notFound))
+            .map { p in
+                // TODO: consider which missing values take defaults and which ones error out
+                // Or do we make view model fields optional and let display deal with it?
+                // If for instance we have just a package entry with no repository, we should
+                // probably error out (because too many fields would be blank).
+                Self.init(title: p.name ?? "–",
+                          url: p.url,
+                          license: p.repository?.license ?? .none,
+                          summary: p.repository?.summary ?? "–",
+                          authors: [],
+                          history: nil,
+                          activity: nil,
+                          products: p.productCounts)
```

Comment:
I'm a bit on the fence whether to make `title` and `summary` non-optional and avoid the `?? "–"` here. I guess it's better to do it here than having to sprinkle it throughout the view code.

On the other hand, we should probably consider some values as technically optional, because of the way we populate the db but effectively implicitly unwrapped because they need to be there.

Should we make this  this initialiser failable and return `EventLoopFuture<Self?>` such that the caller can bail out? I.e. in `PackageController`:

```
    func show(req: Request) throws -> EventLoopFuture<HTML> {
        guard let id = req.parameters.get("id").flatMap(UUID.init(uuidString:)) else {
            return req.eventLoop.future(error: Abort(.notFound))
        }
        return PackageShowView.Model.query(database: req.db, packageId: id)
            .unwrap(or: Abort(.notFound)  // <- bail out if model query fails or returns insufficient data
            .map { PackageShowView($0).document() }
    }
```

This should be rate but not impossible, for instance if a package is added but then not immediately processed, we'll be missing things title and summary.

When we tackle search we should ensure we filter those results out but since we won't know that here, this might be a workable approach.

Reply:
We need to cope with a missing summary. The GitHub description field where this comes from is not required. There are plenty of packages that have no description.

Having a missing name is much trickier. We should always have a name after an analysis run has taken place, so maybe the package page returns a 404 (and is excluded from search results) until an analysis has processed. I think we can use the existing status field for this?

Reply:
Ok, I'll make the initialiser failable and raise a 404 for a missing package name (title) then, ok? (as part of this PR)

Regarding search, that's what I was thinking, too. Just like we have candidates for analysis we'll have candidates for search and freshly added ones just wouldn't be eligible. And mechanism to steer that is the status and processing stage field. (It's going to be slightly more complicated that that, because we bounce between stages when we re-ingest packages but we can overcome that by introducing another stage or so).

I've got this sort of mapped out but that's probably not something I'd want to mix into this PR here.

Reply:
Sounds good 👍